tests/publish.rs

```diff
@@ -0,0 +1,120 @@
+async fn test_publish_and_subscribe() {
+    test_lib::before_test();
+
+    // Initialization
+    let target_data = String::from("Hello World!");
+    let upclient = UPClientZenoh::new(Config::default()).await.unwrap();
+    let uuri = test_lib::create_utransport_uuri(0);
+
+    // Register the listener
+    let uuri_cloned = uuri.clone();
+    let data_cloned = target_data.clone();
+    let listener = move |result: Result<UMessage, UStatus>| match result {
+        Ok(msg) => {
+            if let Data::Value(v) = msg.payload.unwrap().data.unwrap() {
+                let value = v.into_iter().map(|c| c as char).collect::<String>();
+                assert_eq!(msg.attributes.unwrap().source.unwrap(), uuri_cloned);
+                assert_eq!(value, data_cloned);
+            } else {
+                panic!("The message should be Data::Value type.");
+            }
+        }
+        Err(ustatus) => panic!("Internal Error: {ustatus:?}"),
+    };
+    let listener_string = upclient
+        .register_listener(uuri.clone(), Box::new(listener))
+        .await
+        .unwrap();
+
+    // Send UMessage
+    let umessage = UMessageBuilder::publish(uuri.clone())
+        .with_message_id(UUIDBuilder::new().build())
+        .build_with_payload(
+            target_data.as_bytes().to_vec().into(),
+            UPayloadFormat::UPAYLOAD_FORMAT_TEXT,
+        )
+        .unwrap();
+    upclient.send(umessage).await.unwrap();
+
+    // Waiting for the subscriber to receive data
+    task::sleep(time::Duration::from_millis(1000)).await;
+
+    // Cleanup
+    upclient
+        .unregister_listener(uuri.clone(), &listener_string)
+        .await
+        .unwrap();
+}
+
+#[async_std::test]
+async fn test_notification_and_subscribe() {
+    test_lib::before_test();
+
+    // Initialization
+    let target_data = String::from("Hello World!");
+    let upclient = UPClientZenoh::new(Config::default()).await.unwrap();
+    let src_uuri = test_lib::create_utransport_uuri(0);
+    let dst_uuri = test_lib::create_utransport_uuri(1);
+
+    // Register the listener
+    let uuri_cloned = dst_uuri.clone();
+    let data_cloned = target_data.clone();
+    let listener = move |result: Result<UMessage, UStatus>| match result {
+        Ok(msg) => {
+            if let Data::Value(v) = msg.payload.unwrap().data.unwrap() {
+                let value = v.into_iter().map(|c| c as char).collect::<String>();
+                assert_eq!(msg.attributes.unwrap().sink.unwrap(), uuri_cloned);
+                assert_eq!(value, data_cloned);
+            } else {
+                panic!("The message should be Data::Value type.");
+            }
+        }
+        Err(ustatus) => panic!("Internal Error: {ustatus:?}"),
+    };
+    let listener_string = upclient
+        .register_listener(dst_uuri.clone(), Box::new(listener))
+        .await
+        .unwrap();
+
+    // Send UMessage
+    // TODO: up_rust has bugs while creating UMessageBuilder::notification
```

Comment:
PR into `up_rust` possible? :slightly_smiling_face:

Reply:
Should be fixed [here](https://github.com/eclipse-uprotocol/up-rust/pull/75)